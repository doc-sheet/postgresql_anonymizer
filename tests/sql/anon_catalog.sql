BEGIN;

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

SELECT anon.age(timestamp '2001-04-10', timestamp '1957-06-13') = '43 years 9 mons 27 days';

SELECT pg_typeof(anon.age(timestamp '2001-04-10')) = 'interval'::REGTYPE;

SELECT anon.concat('foo', 'bar') = 'foobar';

SELECT anon.concat('foo', 'bar', 'baz') = 'foobarbaz';

SELECT anon.date_add('2020-03-19 12:00:00-00:00'::timestamp, '1 week') = '2020-03-26 12:00:00-00:00'::timestamp;

SELECT anon.date_part('day', '2020-03-19 12:00:00-00:00'::timestamp) = 19;

SELECT anon.date_subtract('2020-03-19 12:00:00-00:00'::timestamp, '1 week') = '2020-03-12 12:00:00-00:00'::timestamp;

SELECT anon.date_trunc('month', '2020-03-19 12:00:00-00:00'::timestamp) = '2020-03-01 00:00:00-00:00'::timestamp;

SELECT anon.left('foo', 2) = 'fo';

SELECT anon.length('foo') = 3;

SELECT anon.lower('fOO bAr BAz 123') = 'foo bar baz 123';

SELECT anon.make_date(2020, 3, 19) = '2020-03-19'::date;

SELECT anon.make_time(12, 31, 35.08) = '12:31:35.08'::time;

SELECT anon.md5('foo') = 'acbd18db4cc2f85cedef654fccc4a4d8';

SELECT pg_typeof(anon.now()) = 'timestamp with time zone'::REGTYPE;

SELECT anon.right('foo', 2) = 'oo';

SELECT anon.substr('foo', 2) = 'oo';

SELECT anon.substr('bazel', 3, 2) = 'ze';

SELECT anon.to_char(timestamp '2002-04-20 17:31:12.66', 'HH12:MI:SS') = '05:31:12';

SELECT anon.to_char(interval '15h 2m 12s', 'HH24:MI:SS') = '15:02:12';

SELECT anon.to_char(125, '999') = ' 125';

SELECT anon.to_char(125.8, '999D9') = ' 125.8';

SELECT anon.to_char(-125.8, '999D99S') = '125.80-';

SELECT anon.to_date('05 Dec 2000', 'DD Mon YYYY') = '2000-12-05'::DATE;

SELECT anon.to_number('12,454.8-', '99G999D9S') = -12454.8;

SELECT pg_typeof(anon.to_timestamp('05 Dec 2000', 'DD Mon YYYY')) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;

SELECT anon.upper('fOO bAr BAz 123') = 'FOO BAR BAZ 123';

ROLLBACK;
