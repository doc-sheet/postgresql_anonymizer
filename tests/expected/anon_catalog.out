BEGIN;
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
SELECT anon.age(timestamp '2001-04-10', timestamp '1957-06-13') = '43 years 9 mons 27 days';
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.age(timestamp '2001-04-10')) = 'interval'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.concat('foo', 'bar') = 'foobar';
 ?column? 
----------
 t
(1 row)

SELECT anon.concat('foo', 'bar', 'baz') = 'foobarbaz';
 ?column? 
----------
 t
(1 row)

SELECT anon.date_add('2020-03-19 12:00:00-00:00'::timestamp, '1 week') = '2020-03-26 12:00:00-00:00'::timestamp;
 ?column? 
----------
 t
(1 row)

SELECT anon.date_part('day', '2020-03-19 12:00:00-00:00'::timestamp) = 19;
 ?column? 
----------
 t
(1 row)

SELECT anon.date_subtract('2020-03-19 12:00:00-00:00'::timestamp, '1 week') = '2020-03-12 12:00:00-00:00'::timestamp;
 ?column? 
----------
 t
(1 row)

SELECT anon.date_trunc('month', '2020-03-19 12:00:00-00:00'::timestamp) = '2020-03-01 00:00:00-00:00'::timestamp;
 ?column? 
----------
 t
(1 row)

SELECT anon.left('foo', 2) = 'fo';
 ?column? 
----------
 t
(1 row)

SELECT anon.length('foo') = 3;
 ?column? 
----------
 t
(1 row)

SELECT anon.lower('fOO bAr BAz 123') = 'foo bar baz 123';
 ?column? 
----------
 t
(1 row)

SELECT anon.make_date(2020, 3, 19) = '2020-03-19'::date;
 ?column? 
----------
 t
(1 row)

SELECT anon.make_time(12, 31, 35.08) = '12:31:35.08'::time;
 ?column? 
----------
 t
(1 row)

SELECT anon.md5('foo') = 'acbd18db4cc2f85cedef654fccc4a4d8';
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.now()) = 'timestamp with time zone'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.right('foo', 2) = 'oo';
 ?column? 
----------
 t
(1 row)

SELECT anon.substr('foo', 2) = 'oo';
 ?column? 
----------
 t
(1 row)

SELECT anon.substr('bazel', 3, 2) = 'ze';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_char(timestamp '2002-04-20 17:31:12.66', 'HH12:MI:SS') = '05:31:12';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_char(interval '15h 2m 12s', 'HH24:MI:SS') = '15:02:12';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_char(125, '999') = ' 125';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_char(125.8, '999D9') = ' 125.8';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_char(-125.8, '999D99S') = '125.80-';
 ?column? 
----------
 t
(1 row)

SELECT anon.to_date('05 Dec 2000', 'DD Mon YYYY') = '2000-12-05'::DATE;
 ?column? 
----------
 t
(1 row)

SELECT anon.to_number('12,454.8-', '99G999D9S') = -12454.8;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.to_timestamp('05 Dec 2000', 'DD Mon YYYY')) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.upper('fOO bAr BAz 123') = 'FOO BAR BAZ 123';
 ?column? 
----------
 t
(1 row)

ROLLBACK;
